src/vm/object.cpp

```diff
@@ -1525,21 +1525,21 @@ void STDCALL CopyValueClassUnchecked(void* dest, void* src, MethodTable *pMT)
     switch (pMT->GetNumInstanceFieldBytes())
     {        
     case 1:
-        VolatileStore((UINT8*)dest, *(UINT8*)src);
+        *(UINT8*)dest = *(UINT8*)src;
         break;
 #ifndef ALIGN_ACCESS
         // we can hit an alignment fault if the value type has multiple 
         // smaller fields.  Example: if there are two I4 fields, the 
         // value class can be aligned to 4-byte boundaries, yet the 
         // NumInstanceFieldBytes is 8
```

Comment:
I was hitting these asserts in unboxing. Barriers are pretty heavy weight for that use case.  

Is there a way to tell when we are being called by reflection API?  Is there a JIT Helper for reflection?

I can add an optional isUnaligned parameter to VolatileStore...